docs/release-checklist.rst

```diff
@@ -0,0 +1,155 @@
+- Push the branch to github
+- Create a (draft) PR; this should trigger CI (note that github
+  doesn't let you create a PR without some changes on the branch so
+  running + commiting the NEWS.txt file achieves that without changing
+  any code)
+- Confirm CI runs successfully on all platforms
+
+
+Create Release Candidate
+````````````````````````
+
+Before "officially" tagging any release, we will make a
+release-candidate available. So there will be at least 1.15.0rc0 (for
+example). If there are any problems, an rc1 or rc2 etc may also be
+released. Anyone can sign these releases (ideally they'd be signed
+"officially" as well, but it's better to get them out than to wait for
+that).
+
+Typically expert users will be the ones testing release candidates and
+they will need to evaluate which contributers' signatures they trust.
+
+- (all steps above are completed)
+- sign the release
+  - git tag -s -u 0xE34E62D06D0E69CFCA4179FFBDE0D31D68666A7A -m "release Tahoe-LAFS-1.15.0rc0" tahoe-lafs-1.15.0.rc0
+  - (replace the key-id above with your own)
+- build all code locally
+  - these should all pass:
+    - tox -e py27,codechecks,docs,integration
+  - these can fail (ideally they should not of course):
+    - tox -e deprecations,upcoming-deprecations
+- build tarballs
+  - tox -e tarballs
+  - confirm it at least exists:
+  - ls dist/ | grep 1.15.0rc0
+- inspect and test the tarballs
+  - install each in a fresh virtualenv
+  - run basic tests
+- when satisfied, sign the tarballs:
+  - gpg --pinentry=loopback --armor --sign dist/tahoe_lafs-1.15.0rc0-py2-none-any.whl
+  - gpg --pinentry=loopback --armor --sign dist/tahoe_lafs-1.15.0rc0.tar.bz2
+  - gpg --pinentry=loopback --armor --sign dist/tahoe_lafs-1.15.0rc0.tar.gz
+  - gpg --pinentry=loopback --armor --sign dist/tahoe_lafs-1.15.0rc0.zip
+
+
+Privileged Contributor
+-----------------------
+
+Steps in this portion require special access to keys or
+infrastructure. For example, **access to tahoe-lafs.org** to upload
+binaries or edit HTML.
+
+
+Hack Tahoe-LAFS
+```````````````
+
+Did anyone contribute a hack since the last release? If so, then
+https://tahoe-lafs.org/hacktahoelafs/ needs to be updated.
+
+
+Upload Artifacts
+````````````````
+
+Any release-candidate or actual release plus signature (.asc file)
+need to be uploaded to https://tahoe-lafs.org in ~source/downloads
+
+- secure-copy all release artifacts to the download area on the
+  tahoe-lafs.org host machine. `~source/downloads` on there maps to
+  https://tahoe-lafs.org/downloads/ on the Web.
+- scp dist/*1.15.0* meejah@tahoe-lafs.org:/home/source/downloads
+- the following developers have access to do this:
+  - exarkun
+  - meejah
+  - warner
+
+For the actual release, the tarball and signature files need to be
+uploaded to PyPI as well.
+
+- how to do this?
+- (original guide says only "twine upload dist/*")
+- the following developers have access to do this:
+  - exarkun
+  - warner
```

Comment:
I wonder if I do.  PyPI says you, warner, and I are "maintainers" on the PyPI project.  I don't know what the "maintainer" permission confers.  I can't find any kind of administrative interface for the project so I can't see what permissions anyone has or grant any new permissions.

Reply:
What I learned last week is that a PyPI project maintainer can upload packages, but they cannot "manage" things (which includes permission to yank/delete a release or delete some packages selectively).  You have to be a project "owner" in order to be able to yank/delete stuff.

@vu3rdd made me a zfec project "owner", which is how I could delete those broken .whl packages.